core/src/navigation/TopNav.html

```diff
@@ -11,135 +11,162 @@
-            >
-              <nav class="fd-menu">
-                <ul class="fd-menu__list">
-                  {#each children as node} {#if !node.hideFromNav}
-                  <li>
-                    <a
-                      class="fd-menu__item {node === selectedNode ? 'is-selected' : ''} {node.externalLink && node.externalLink.url ? 'fd-global-nav__btn--external' : ''}"
-                      on:click="handleClick(node)"
-                    >
-                      {#if node.icon} {#if hasOpenUIicon(node)}
-                      <span
-                        class="fd-top-nav__icon sap-icon--{node.icon} sap-icon--m"
-                      ></span>
-                      {:else}
-                      <img
-                        class="fd-top-nav__icon nav-icon"
-                        src="{node.icon}"
+              <div
+                class="fd-popover__body fd-popover__body--right"
+                aria-hidden="{!(dropDownStates.overflowPopover || false)}"
+                id="overflowPopover"
+              >
+                <nav class="fd-menu">
+                  <ul class="fd-menu__list">
+                    {#each children as node} {#if !node.hideFromNav}
+                    <li>
+                      <a
+                        class="fd-menu__item {node === selectedNode ? 'is-selected' : ''} {node.externalLink && node.externalLink.url ? 'fd-global-nav__btn--external' : ''}"
+                        on:click="handleClick(node)"
                       >
-                      {/if} {:else} {node.label} {/if} {#if node.externalLink &&
-                      node.externalLink.url}
-                      <span
-                        class="sap-icon--action sap-icon--s"
-                      ></span>
-                      {/if}
-                    </a>
-                  </li>
-                  {/if} {/each}
-                </ul>
-              </nav>
+                        {#if node.icon} {#if hasOpenUIicon(node)}
+                        <span
+                          class="fd-top-nav__icon sap-icon--{node.icon} sap-icon--m"
+                        ></span>
+                        {:else}
+                        <img
+                          class="fd-top-nav__icon nav-icon"
+                          src="{node.icon}"
+                        >
+                        {/if} {:else} {node.label} {/if} {#if node.externalLink &&
+                        node.externalLink.url}
+                        <span
+                          class="sap-icon--action sap-icon--s"
+                        ></span>
+                        {/if}
+                      </a>
+                    </li>
+                    {/if} 
+                    {/each}
+                    {#if isProductSwitcherAvailable}
+                      <li>       
+                        <a class="fd-menu__item" on:click="openMobileProductSwitcher(event)"  data-cy="mobile-product-switcher">            
```

Comment:
Now we could just leave for the click event `toggleDropdownState('productSwitcherPopover')`, nothing else is done in the `openMobileProductSwitcher` function anymore.

Reply:
You are wrong, the event propagation is stopped. That's why it doesn't work when I put `toggleDropdownState` here. But I did this change in `ProductSwitcher.html` where stopping propagation wasn't needed :)